src/scripts/setNetwork.js

```diff
@@ -20,11 +20,33 @@ export const setNetwork = async ({ to, next }, apollo, store) => {
   if (path == "/") {
     path = "/portfolio"
   }
+
+  // if the url is of the form /cosmos-hub/portfolio the path we want to add later is /portfolio and not the whole thing
+  if (to.params.networkId) {
+    const subPathRegexp = /\/.+(\/.+)/ //extracts the subpath
+    const match = subPathRegexp.exec(to.path)
+    if (match && match.length >= 2) {
+      // the path is is of form /cosmos-hub/portfolio
+      path = match[1]
+    } else {
+      // the path is is of form /cosmos-hub so we forward to portfolio
+      path = "/portfolio"
+    }
```

Comment:
There is something here I don't get. Wasn't the reloading happening in production?

Or it was also happening here? In what cases? I didn't notice 👀